lib/src/back_end/code_writer.dart

```diff
@@ -17,15 +17,11 @@ import 'solution.dart';
 /// an instance of this class. It has methods that the piece can call to add
 /// output text to the resulting code, recursively format child pieces, insert
 /// whitespace, etc.
-///
-/// This class also accumulates the score (the relative desireability of a set
-/// of formatting choices) that the resulting code has by tracking things like
-/// how many characters of code overflow the page width.
 class CodeWriter {
   final int _pageWidth;
 
-  /// The state values for the pieces being written.
-  final PieceStateSet _pieceStates;
+  /// The solution being formatted.
```

Comment:
[nit] I would expect a "solution" to already have a formatting. If the formatting hasn't been solved yet - what does "solution" refer to?

Reply:
Reworded and added some docs elsewhere.

I would characterize it like:

* Solution has all of the state required to describe a complete formatting solution. The primary state is the set of States bound to various Pieces. Then the total overflow characters, cost, and output text are derived from that state. They are mutable state, but only mutated by CodeWriter. Once CodeWriter has finished visiting the entire piece tree, Solution becomes effectively immutable.
* CodeWriter is the interface Pieces use to implement their formatting logic. It also stores any emphemeral state used while in the process of generating the formatted text from a given set of piece states but which isn't needed once we know the final text, cost, and overflow.